sdks/python/apache_beam/transforms/display/display_data.py

```diff
@@ -0,0 +1,133 @@
+
+  def display_data(self):
+    return {}
+
+  def _namespace(self):
+    return '{}.{}'.format(self.__module__, self.__class__.__name__)
+
+
+class DisplayData(object):
+  def __init__(self, namespace='__main__'):
+    self.namespace = namespace
+    self.items = []
+
+  def populate_items(self, display_data_dict):
+    for key, element in display_data_dict.items():
+      if isinstance(element, HasDisplayData):
+        subcomponent_display_data = DisplayData(element._namespace())
+        subcomponent_display_data.populate_items(element.display_data())
+        self.items += subcomponent_display_data.items
+        continue
+
+      if isinstance(element, dict):
+        self.items.append(
+            DisplayDataItem(self.namespace,
+                            key,
+                            DisplayDataItem._get_value_type(element['value']),
+                            element['value'],
+                            shortValue=element.get('shortValue'),
+                            url=element.get('url'),
+                            label=element.get('label')))
+        continue
+
+      # If it's not a HasDisplayData element,
+      # nor a dictionary, then it's a simple value
+      self.items.append(
+          DisplayDataItem(self.namespace,
+                          key,
+                          DisplayDataItem._get_value_type(element),
+                          element))
+
+  def output(self):
+    return [item.get_dict() for item in self.items]
+
+  @classmethod
+  def create_from(cls, has_display_data):
+    if not isinstance(has_display_data, HasDisplayData):
+      raise ValueError('Element of class {}.{} does not subclass HasDisplayData'
+                       .format(has_display_data.__module__,
+                               has_display_data.__class__.__name__))
+    display_data = cls(has_display_data._namespace())
+    display_data.populate_items(has_display_data.display_data())
+    return display_data
+
+
+class DisplayDataItem(object):
+  typeDict = {str:'STRING',
+              int:'INTEGER',
+              float:'FLOAT',
+              timedelta:'DURATION',
+              datetime:'TIMESTAMP'}
+
+  def __init__(self, namespace, key, type_, value,
+               shortValue=None, url=None, label=None):
+    if key is None:
+      raise ValueError('Key must not be None')
+    if value is None:
+      raise ValueError('Value must not be None')
+    if type_ is None:
+      raise ValueError('Value {} is of an unsupported type.'.format(value))
+
+    self.namespace = namespace
+    self.key = key
+    self.type = type_
+    self.value = value
+    self.shortValue = shortValue
+    self.url = url
+    self.label = label
+
+  def get_dict(self):
+    res = {'key': self.key,
+           'namespace': self.namespace,
+           'type': self.type}
+
+    if self.url is not None:
+      res['url'] = self.url
+    # TODO: What to do about shortValue? No special processing?
```

Comment:
In java, shortValue is currently only used for a compressed version of `JAVA_CLASS`, displaying the class name without module.